extensions/podman/packages/extension/src/extension.spec.ts

```diff
@@ -750,6 +752,95 @@ describe.each([
   });
 });
 
+test.each([
+  { architecture: 'arm64', expectedProvider: VMTYPE.LIBKRUN },
+  { architecture: 'x64', expectedProvider: VMTYPE.APPLEHV },
+])('verify create on mac from settings on %s', async ({ architecture, expectedProvider }) => {
+  vi.mocked(extensionApi.env).isMac = true;
+  vi.mocked(arch).mockReturnValue(architecture);
+  vi.mocked(extensionApi.process.exec).mockResolvedValueOnce({
+    stdout: `podman version 5.4.0`,
+  } as extensionApi.RunResult);
+
+  await extension.createMachine(
+    {
+      'podman.factory.machine.cpus': '2',
+      'podman.factory.machine.image': 'path',
+      'podman.factory.machine.memory': '1048000000', // 1048MB = 999.45MiB
+      'podman.factory.machine.diskSize': '250000000000', // 250GB = 232.83GiB
+      'podman.factory.machine.provider': expectedProvider,
```

Comment:
this is the only line that differs within these two parametrized test cases. Can we make it a function and just change the argument?